src/Icicle/Avalanche/Statement/Simp/Constructor.hs

```diff
@@ -96,28 +127,51 @@ constructor a_fresh statements
    , Just (_,_,_,_,b) <- fromSum x'
    = b
 
-   | Just (PrimUnsafe (PrimUnsafeArrayIndex _), [XVar _ n, ix]) <- takePrimApps x
-   , Just x' <- get env n
-   , Just (ta, tb, a, b) <- fromZippedArray x'
+   -- update
+   | Just (PrimUpdate (PrimUpdateArrayPut _), [arr,ix,v]) <- takePrimApps x
+   , Just arrx              <- resolve env arr
+   , Just (ta, tb, i, a, b) <- fromSummedArray arrx
+   , Just vx                <- resolve env v
+   , Just (_,_,boool,l,r)   <- fromSum vx
+   , tai                    <- ArrayT BoolT
+   , taa                    <- ArrayT ta
+   , tab                    <- ArrayT tb
+   = case boool of
+      XValue _ _ (VBool False)
```

Comment:
oh right, that was stupid, what was I thinking =( eh, maybe this simplification isn't necessary, but then it won't get rid of unsum/unzip